language-server/dm/completionSuggester.ml

```diff
@@ -26,11 +105,27 @@ let get_hyps st loc =
   DocumentManager.get_proof st loc
     |> Option.map (fun Proof.{ goals; sigma; _ } -> Option.cata (mk_hyps sigma) [] (List.nth_opt goals 0)) 
  
+let take n l =
+  let rec sub_list n accu l =
+    match l with 
+    | [] -> accu 
+    | hd :: tl ->
+      if n = 0 then accu 
+      else sub_list (n - 1) (hd :: accu) tl
+  in
+  List.rev (sub_list n [] l)
+
 let get_completion_items ~id params st loc =
   let open Yojson.Basic.Util in
   let hypotheses = get_hyps st loc in
   let lemmasOption = DocumentManager.get_lemmas st loc in
-  let lemmas = lemmasOption |> Option.map (List.map CompletionItems.pp_completion_item) in
-  [hypotheses; lemmas] 
+  let goal, sigma, env = get_goal_type st loc in
+  let lemmas = lemmasOption |> Option.map 
+    (fun l -> 
+      rank_choices goal sigma env l |> 
+      take 10000 |> 
```

Comment:
Did you try if vscode can really handle that many results? (even faking the list).
Cause it looks like *a lot* to me.

Reply:
It might be a bit excessive, I just set it to 10k so that our `sortText` works right now, as we are using 5 digit numbers to sort the results with. 
We will figure out a more reasonable limit at some point.